### PR TITLE
RSS-K8: Adding scala plugin to pom file to be able to run scala tests, as wel…

### PR DESCRIPTION
**Pull request by mabansal**
Currently, K8 branch does not run with scala tests, add the scala plugin in the pom file to be able to run Scala tests
, as well as fixing java code for not mutating variables

### CODE REVIEW

**Review by vectorijk — APPROVED** [review]

**Review by mayurdb — APPROVED** [review]

**Comment by mayurdb** [comment]
lgtm, we are already using these dependencies internally to run scala UTs

File: pom.xml [hunk 1]
```diff
@@ -346,6 +346,52 @@
...
+                        <goals>
+                            <goal>testCompile</goal>
+                        </goals>
+                    </execution>
+                    <!--<execution>
+                        <id>attach-javadocs</id>
+                        <goals>
+                            <goal>doc-jar</goal>
+                        </goals>
+                    </execution>-->
```
Comment (vectorijk) [inline pom.xml:374]: let's remove this unused execution step in scala maven plugin setting .

Comment (mabansal) [reply pom.xml:374]: Done

**Comment by mabansal** [comment]
> lgtm, we are already using these dependencies internally to run scala UTs

Yup you are correct, we have to add them in K8 branch as well as main branch, will add them in main branch in a seperate PR

**Comment by mabansal** [comment]
merging in

**Pull request merged**